integreat_cms/static/src/js/pois/poi-actions.ts

```diff
@@ -0,0 +1,93 @@
+import { getCsrfToken } from "../utils/csrf-token";
+
+window.addEventListener("load", () => {
+  document.getElementById("id_address")?.addEventListener("focusout", autoCompleteAddress);
+  document.getElementById("id_postcode")?.addEventListener("focusout", autoCompleteAddress);
+  document.getElementById("id_city")?.addEventListener("focusout", autoCompleteAddress);
+
+  let longitude = document.getElementById("id_longitude") as HTMLInputElement;
+  let latitude = document.getElementById("id_latitude") as HTMLInputElement;
+
+  document.getElementById("auto-fill-coordinates")?.addEventListener("input", ({ target }) => {
+    if ((target as HTMLInputElement).checked) {
+      autoCompleteAddress();
+    } else {
+      // Reset to initial value
+      longitude.value = longitude.dataset.initial;
+      latitude.value = latitude.dataset.initial;
+    }
+  });
+
+  // Add possibility to reset coordinates to initial values
+  if (longitude?.value) {
+    longitude.dataset.initial = longitude.value;
```

Comment:
I'd store the initial values in the data attributes rather than in global variables.